### PR TITLE
Implementing Broken Horn, Vizago's Pride

### PR DESCRIPTION
**Pull request by aruizc**
image

### CODE REVIEW

File: test/server/cards/07_LAW/units/BrokenHornVizagosPride.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+describe('Broken Horn, Vizago\'s Pride', function () {
+    integration(function (contextRef) {
+        describe('Broken Horn, Vizago\'s Pride when played ability', function () {
+            it('should let the player to draw a card and resource the top card of their deck', async function () {
```
Comment (VJubert) [inline test/server/cards/07_LAW/units/BrokenHornVizagosPride.spec.ts]: can you add a test with an empty deck. It should take 3 damage from draw and not resource anything

File: server/game/cards/07_LAW/units/BrokenHornVizagosPride.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        return {
+            id: 'broken-horn#vizagos-pride-id',
+            internalName: 'broken-horn#vizagos-pride'
+        };
+    }
+
+    public override setupCardAbilities(registrar: INonLeaderUnitAbilityRegistrar, AbilityHelper: IAbilityHelper) {
+        registrar.addWhenPlayedAbility({
+            title: 'If you have fewer cards in hand than your opponent, draw a card. If you control fewer resources than your opponent, resource the top card of your deck',
+            immediateEffect: AbilityHelper.immediateEffects.sequential([
```
Comment (AMMayberry1) [inline server/game/cards/07_LAW/units/BrokenHornVizagosPride.ts]: This should be `simultaneous()`

File: test/server/cards/07_LAW/units/BrokenHornVizagosPride.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                expect(context.greenSquadronAwing).toBeInZone('deck');
+                context.player1.clickCard(context.brokenHornVizagosPride);
+
+                // Player 1 should draw a card and resource the top card of their deck since they have fewer cards in hand and fewer resources than Player 2
+                expect(context.player1.hand.length).toBe(startingHandSize); // As player drawed a card but also played a card
+                expect(context.player1.resources.length).toBe(startingResourcesP1 + 1);
+                expect(context.greenSquadronAwing).toBeInZone('hand', context.player1);
+                expect(context.wampa).toBeInZone('resource', context.player1);
+                expect(context.player2).toBeActivePlayer();
+            });
```
Comment (AMMayberry1) [inline test/server/cards/07_LAW/units/BrokenHornVizagosPride.spec.ts]: Let's add a test where there is one card left in deck, then confirm that it goes to hand and nothing is resourced

File: server/game/cards/07_LAW/units/BrokenHornVizagosPride.ts (hunk 2 above)
Comment (aruizc) [reply server/game/cards/07_LAW/units/BrokenHornVizagosPride.ts]: I tried using simultaneous(), but it looks like draw and resource are affecting the same card at the same time. I think this should be sequential: first draw the card, then resource it. Let me know your thoughts
